src/P2P/Node.hs

```diff
@@ -345,15 +345,12 @@ guardPeerDb
 guardPeerDb v nid peerDb pinf = do
     peers <- peerDbSnapshot peerDb
     if
-        | isKnown peers -> return $ Right pinf
+        | isKnown peers pinf -> return $ Right pinf
```

Comment:
When called via `syncFromPeer` this is redundant, but this function is also called from elsewhere so we leave this check in here.